### PR TITLE
[WIP] #1802: The "three dots" context menu is not closed after unsuccessful licensing from Media Gallery

### LINKED ISSUE #1802: The "three dots" context menu is not closed after unsuccessful licensing from Media Gallery
### Preconditions (*)
1. Install Magento with Adobe Stock Integration
2. Configured integration in Stores -> Configuration -> Advanced-> System -> Adobe Stock Integration fieldset
3. Old Media Gallery disabled in Stores -> Configuration -> Advanced-> System -> Media Gallery -> Enable Old Media Gallery - No
4. Provided _API Key (Client ID)_ . **Not** provided _Client Secret_
client_key
5. Unlicensed image preview saved to Media Gallery being at the bottom of images

### Steps to reproduce (*)
1. Go to **Content - Media Gallery**, click **Search Adobe Stock**
2. Select the previously saved Unlicensed image
3. Click "three dots" and select **License**
4. Do not click anywhere, observe the behavior

### Expected result (*)
The context menu is closed after _Login failed..._ message appears 
contextmenu_expected

### Actual result (*)
The context menu is NOT closed after _Login failed..._ message appears
contextmenu
context

### PR DESCRIPTION
**Pull request by joweecaquicla**
### Description (*)

This PR will introduce the MFTF test for the fix on https://github.com/magento/adobe-stock-integration/issues/1802. Link of the fix's PR: https://github.com/magento/magento2/pull/29967
### Fixed Issues (if relevant)

1. Fixes magento/adobe-stock-integration https://github.com/magento/adobe-stock-integration/issues/1802: The "three dots" context menu is not closed after unsuccessful licensing from Media Gallery
2. ...

### Manual testing scenarios (*)

1. ...
2. ...

### CODE REVIEW

**Comment by joweecaquicla** [comment]
@magento run Functional Tests EE, Functional Tests CE, Functional Tests B2B

File: AdobeStockImageAdminUi/Test/Mftf/Section/AdminEnhancedMediaGalleryImageActionsSection.xml [hunk 1]
```diff
@@ -9,5 +9,6 @@
 	xsi:noNamespaceSchemaLocation="urn:magento:mftf:Page/etc/SectionObject.xsd">
     <section name="AdminEnhancedMediaGalleryImageActionsSection">
         <element name="license" type="button" selector="[data-ui-id='action-license']"/>
+        <element name="contextMenu" type="block" selector="//div[@class='media-gallery-image']//ul[@class='action-menu _active']//li//a[@class='action-menu-item']"/>
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Section/AdminEnhancedMediaGalleryImageActionsSection.xml]: I'd rename this element to the `contextMenuItem` and move it to MediaGalleryUi (as this is the module introducing this menu)

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminStandaloneMediaGalleryUnsuccessfulLicensingTest.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        <actionGroup ref="AdminMediaGalleryFolderSelectActionGroup" stepKey="selectTestFolderToOpen">
+            <argument name="name" value="testFolder"/>
+        </actionGroup>
+        <click selector="{{AdminEnhancedMediaGalleryImageActionsSection.openContextMenu}}" stepKey="openContextMenu"/>
+        <click selector="{{AdminEnhancedMediaGalleryImageActionsSection.license}}" stepKey="licenseImage"/>
+        <waitForLoadingMaskToDisappear stepKey="waitForLoadingMaskToDisappear"/>
+        <actionGroup ref="AdminAdobeStockImsPopupSignInFillUserDataActionGroup" stepKey="fillUserCredentials"/>
+        <actionGroup ref="AdminAdobeStockImsPopupClickSignInActionGroup" stepKey="clickSignInImsPopup"/>
+        <waitForElementVisible selector="{{AdminMessagesSection.errorMessage}}" stepKey="waitErrorMessage"/>
+        <dontSeeElement selector="{{AdminEnhancedMediaGalleryImageActionsSection.contextMenu}}" stepKey="verifyContextMenuIsClosed" />
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminStandaloneMediaGalleryUnsuccessfulLicensingTest.xml]: Let's extract this check to AssetAdminMediaGalleryContextMenuOpenedGroup (and place it in the MediaGalleryUi module)

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminStandaloneMediaGalleryUnsuccessfulLicensingTest.xml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            <stories value="User unsuccessful licensing of image in Media Gallery grid"/>
+            <testCaseId value="https://studio.cucumber.io/projects/131313/test-plan/folders/1054245/scenarios/3586914"/>
+            <title value="User unsuccessful licensing of image in Media Gallery grid"/>
+            <description value="User unsuccessful licensing of image in Media Gallery grid"/>
+            <severity value="CRITICAL"/>
+            <group value="adobe_stock_media_gallery"/>
+        </annotations>
+        <before>
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <actionGroup ref="AdminAdobeStockSetConfigActionGroup" stepKey="setIncorrectAdobeSecret">
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminStandaloneMediaGalleryUnsuccessfulLicensingTest.xml:27]: Please revert the configuration changes in `after` section

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminStandaloneMediaGalleryUnsuccessfulLicensingTest.xml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            <actionGroup ref="AdminMediaGalleryOpenNewFolderFormActionGroup" stepKey="openNewFolderForm"/>
+            <actionGroup ref="AdminMediaGalleryCreateNewFolderActionGroup" stepKey="createTestFolder">
+                <argument name="name" value="testFolder"/>
+            </actionGroup>
+            <actionGroup ref="AdminMediaGalleryAssertFolderNameActionGroup" stepKey="assertTestFolderCreated">
+                <argument name="name" value="testFolder"/>
+            </actionGroup>
+            <waitForPageLoad stepKey="waitForGridToLoadAfterTestFolderCreated"/>
+            <actionGroup ref="AdminEnhancedMediaGallerySearchAdobeStockActionGroup" stepKey="openAdobeStockGrid"/>
+            <actionGroup ref="AdminSearchImagesOnModalActionGroup" stepKey="searchForUnlicensedImage">
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminStandaloneMediaGalleryUnsuccessfulLicensingTest.xml:41]: Please reset the filters applied to adobe stock grid in `after` section

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminStandaloneMediaGalleryUnsuccessfulLicensingTest.xml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        </after>
+        <actionGroup ref="AdminOpenStandaloneMediaGalleryActionGroup" stepKey="openMediaGallery"/>
+        <actionGroup ref="ResetAdminDataGridToDefaultViewActionGroup" stepKey="resetAdminDataGridToDefaultView"/>
+        <actionGroup ref="AdminMediaGalleryFolderSelectActionGroup" stepKey="selectTestFolderToOpen">
+            <argument name="name" value="testFolder"/>
+        </actionGroup>
+        <click selector="{{AdminEnhancedMediaGalleryImageActionsSection.openContextMenu}}" stepKey="openContextMenu"/>
+        <click selector="{{AdminEnhancedMediaGalleryImageActionsSection.license}}" stepKey="licenseImage"/>
+        <waitForLoadingMaskToDisappear stepKey="waitForLoadingMaskToDisappear"/>
+        <actionGroup ref="AdminAdobeStockImsPopupSignInFillUserDataActionGroup" stepKey="fillUserCredentials"/>
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminStandaloneMediaGalleryUnsuccessfulLicensingTest.xml:72]: This test might need to be skipped by default with a link to an issue https://github.com/magento/adobe-stock-integration/issues/1170

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the test @joweecaquicla ! Please see my comments

**Pull request merged**